test/tutorials/calculational_logic.jl

```diff
@@ -13,16 +14,16 @@ include(joinpath(dirname(pathof(Metatheory)), "../examples/calculational_logic_t
   @test @areequal calculational_logic_theory ((!p == !p) == true) true
   @test @areequal calculational_logic_theory ((!p || !p) == !p) (!p || p) !(!p && p) true
   @test @areequal calculational_logic_theory ((p ⟹ (p || p)) == true) true
-  params = SaturationParams(timeout = 12, eclasslimit = 10000, schedulerparams = (1000, 5))
 
+  params = SaturationParams(timeout = 12, eclasslimit = 10000, schedulerparams = (1000, 5))
   @test areequal(calculational_logic_theory, :(((p ⟹ (p || p)) == ((!(p) && q) ⟹ q))), true; params = params)
 
   ex = :((p ⟹ (q ⟹ r)) ⟹ ((p ⟹ q) ⟹ (p ⟹ r)))   # Frege's theorem
-  res = areequal(calculational_logic_theory, ex, true; params = params)
-  @test_broken !ismissing(res) && res
-
+  params = SaturationParams(timeout = 12, eclasslimit = 10000, schedulerparams = (6000, 5))
+  @test prove(calculational_logic_theory, ex, 2, 10, params)
```

Comment:
I could not get this to work with `areequal`, ok with you to use `prove` here? If yes, then should we keep the `prove` function in the examples or maybe move it into MT itself?

Reply:
`@areequal` will be deprecated.

Reply:
`prove` is just a `FixpointNoCycle` combinator and should be defined in terms of it with attention to performance, when we give attention to `Rewriters.jl` module